src/firstboot/firstboot.c

```diff
@@ -1089,6 +1090,48 @@ static int process_kernel_cmdline(int rfd) {
         return 0;
 }
 
+static int reset_one(int rfd, const char *path, ChaseFlags flags) {
+        _cleanup_close_ int pfd = -EBADF;
+        _cleanup_free_ char *f = NULL;
+
+        assert(rfd >= 0);
+        assert(path);
+
+        pfd = chase_and_open_parent_at(rfd, path, CHASE_AT_RESOLVE_IN_ROOT|CHASE_WARN|flags, &f);
+        if (pfd == -ENOENT)
+                return 0;
+        if (pfd < 0)
+                return log_error_errno(pfd, "Failed to resolve %s: %m", path);
+
+        if (unlinkat(pfd, f, 0) < 0)
+                return errno == ENOENT ? 0 : log_error_errno(errno, "Failed to remove %s: %m", path);
+
+        log_info("Removed %s", path);
+        return 0;
+}
+
+static int process_reset(int rfd) {
+        int r;
+
+        assert(rfd >= 0);
+
+        if (!arg_reset)
+                return 0;
+
+        FOREACH_STRING(p,
+                       "/etc/locale.conf",
+                       "/etc/vconsole.conf",
+                       "/etc/hostname",
+                       "/etc/machine-id",
+                       "/etc/kernel/cmdline") {
+                r = reset_one(rfd, p, /* flags = */ 0);
```

Comment:
Here, also CHASE_NOFOLLOW should be used?

Reply:
Then we should also use it when writing the files. It depends on what we want the behavior to be.

Debian is a good example. On Debian, /etc/locale.conf is a symlink to /etc/default/locale. Should firstboot operate on /etc/locale.conf in that case or on /etc/default/locale?

Reply:
I think the answer is to use CHASE_NOFOLLOW in both cases.

An easier example is: let's say `/etc/locale.conf` is a symlink to `/usr/lib/distro-defaults/locale.conf`. In that case, I'd say it's pretty obvious that we shouldn't follow the symlink. But we can't easily distinguish the cases where we can and cannot. `/etc/default/` is an idea that doesn't make much sense… I don't think we need to cater to it.

Reply:
We now use CHASE_NOFOLLOW everywhere.